samples/employee-portal/hr/hrservice.bal

```diff
@@ -45,100 +45,72 @@ service<http:Service> hr bind { port: 8080 } {
-                            }
+        string employeeName = req.getHeader("x-vick-auth-subject");
+        io:println("employee name: " + employeeName);
+
+        json employeeDetails;
+            match getEmployeeDetails(req) {
+                http:Response response => {
+                    match response.getJsonPayload() {
+                        json jsonEmpDetails => {
+                            employeeDetails = jsonEmpDetails;
+                            io:println("employee details: ");
+                            io:println(employeeDetails);
+                        }
+                        error e => {
+                            log:printError("Error in extracting response from Employee service", err = e);
+                            res.statusCode = 500;
+                            caller->respond(res) but { error e1 => log:printError("Error sending response", err = e1) };
+                            done;
                         }
                     }
                 }
                 error e => {
-                    // jwt token extraction error
-                    log:printError("Unabe to extract JWT token", err = e);
-                    res.statusCode = 401;
+                    res.statusCode = 500;
+                    caller->respond(res) but { error e1 => log:printError("Error sending response", err = e1) };
+                    done;
                 }
             }
-        } else {
-            // no authorization header, incorrect request
-            log:printInfo("No auth header sent");
-            res.statusCode = 401;
-        }
-        caller->respond(res) but { error e => log:printError("Error sending response", err = e) };
+
+        json stockOptionDetails;
+            match getStockOptionData(req) {
+                http:Response response => {
+                    match response.getJsonPayload() {
+                        json jsonStocks => {
+                            stockOptionDetails = jsonStocks;
+                            io:println("stock details: ");
+                            io:println(stockOptionDetails);
+                        }
+                        error e => {
+                            log:printError("Error in extracting response from Stock service", err = e);
+                            res.statusCode = 500;
+                            caller->respond(res) but { error e1 => log:printError("Error sending response", err = e1) };
+                            done;
+                        }
+                    }
+                }
+                error e => {
+                    res.statusCode = 500;
+                    caller->respond(res) but { error e1 => log:printError("Error sending response", err = e1) };
+                    done;
+                }
+            }
+        json resp = buildResponse(employeeDetails, stockOptionDetails);
+        io:println("response: ");
+        io:println(resp);
+        res.setJsonPayload(untaint resp);
```

Comment:
Have to return the response to the caller?

Reply:
Yes, seems like the sending part is lost here. Will fix it.